src/meshable_arena.cc

```diff
@@ -525,10 +525,7 @@ void MeshableArena::finalizeMesh(void *keep, void *remove, size_t sz) {
 
   void *ptr = mmap(remove, sz, HL_MMAP_PROTECTION_MASK, kMapShared | MAP_FIXED, _fd, keepOff * kPageSize);
   hard_assert_msg(ptr != MAP_FAILED, "mesh remap failed: %d", errno);
-  freePhys(remove, sz);
 
-  int r = mprotect(remove, sz, PROT_READ | PROT_WRITE);
```

Comment:
great find! -- I didn't realize that the `mmap` resets the page protections (though its obvious in hindsight)